controllers/complianceeventsapi/server.go

```diff
@@ -224,67 +249,123 @@ func splitQueryValue(value string) []string {
 // parseQueryArgs will parse the HTTP request's query arguments and convert them to a usable format for constructing
 // the SQL query. All defaults are set and any invalid query arguments result in an error being returned.
 func parseQueryArgs(queryArgs url.Values) (*queryOptions, error) {
-	parsed := &queryOptions{}
+	parsed := &queryOptions{
+		Direction:    "desc",
+		Page:         1,
+		PerPage:      20,
+		Sort:         []string{"compliance_events.timestamp"},
+		ArrayFilters: map[string][]string{},
+		Filters:      map[string][]string{},
+		NullFilters:  []string{},
+	}
 
-	perPageStr := queryArgs.Get("per_page")
+	for arg := range queryArgs {
+		valid := false
 
-	if perPageStr == "" {
-		parsed.PerPage = 20
-	} else {
-		var err error
+		for _, validQueryArg := range validQueryArgs {
+			if arg == validQueryArg {
+				valid = true
```

Comment:
Is it ok only one `validQueryArg`  is valid? 
```

		valid := true

		for _, validQueryArg := range validQueryArgs {
			if arg == validQueryArg {
				valid = false
```

I guess this is right... Could you explain this